src/Symfony/Component/AssetMapper/Tests/ImportMap/RemotePackageStorageTest.php

```diff
@@ -41,19 +41,24 @@ public function testGetStorageDir()
         $this->assertSame(realpath(self::$writableRoot.'/assets/vendor'), realpath($storage->getStorageDir()));
     }
 
-    public function testSaveThrowsWhenVendorDirectoryIsNotWritable()
+    public function testSaveThrowsWhenFailing()
     {
-        $this->filesystem->mkdir($vendorDir = self::$writableRoot.'/assets/acme/vendor');
-        $this->filesystem->chmod($vendorDir, 0555);
+        $vendorDir = self::$writableRoot.'/assets/acme/vendor';
+        $this->filesystem->mkdir($vendorDir.'/module_specifier');
+        $this->filesystem->touch($vendorDir.'/module_specifier/module_specifier.index.js');
+        $this->filesystem->chmod($vendorDir.'/module_specifier/module_specifier.index.js', 0555);
 
         $storage = new RemotePackageStorage($vendorDir);
         $entry = ImportMapEntry::createRemote('foo', ImportMapType::JS, '/does/not/matter', '1.0.0', 'module_specifier', false);
 
         $this->expectException(\RuntimeException::class);
-        $this->expectExceptionMessage('file_put_contents('.$vendorDir.'/module_specifier/module_specifier.index.js): Failed to open stream: No such file or directory');
-        $storage->save($entry, 'any content');
+        $this->expectExceptionMessage('file_put_contents('.$vendorDir.'/module_specifier/module_specifier.index.js): Failed to open stream: Permission denied');
 
-        $this->filesystem->remove($vendorDir);
+        try {
+            $storage->save($entry, 'any content');
+        } finally {
+            $this->filesystem->chmod($vendorDir.'/module_specifier/module_specifier.index.js', 0777);
```

Comment:
this is the fix: revert the chmod before remove()
/cc @xabbuh @smnandre FYI

Reply:
Well...  💡   that makes sense.